src/app.jsx

```diff
@@ -63,6 +64,9 @@ export const Application = () => {
     const [showHidden, setShowHidden] = useState(false);
     const [history, setHistory] = useState([]);
     const [historyIndex, setHistoryIndex] = useState(0);
+    const [isEditing, setIsEditing] = useState(false);
```

Comment:
TO avoid puttong all these extra state in the app.jsx, you could maybe approach it in different way. Namely, when the user wants to edit the file, just change the URL, to `navigator#/?path=home%252Fkkoukiou%252Ftestdir?edit=myfile.txt`

Then all this logic can fit inside the component that does the edit. So no editFile callback in the app.jsx and all the state variables can be moved as well.


I know that's a big code change for no big gain apart from code readability, but I think with the app.jsx growing so much, we are at the edge of not having a readable app.jsx any more.